src/pages/JournalEntryFormPage/JournalEntryFormPage.tsx

```diff
@@ -53,22 +54,25 @@ const JournalEntryFormPage: React.FC = () => {
     }
 
     useEffect(() => {
+        let foundEntry;
         if (entryId) {
-            const foundEntry = mockEntries.find(entry => entry.id === parseInt(entryId)) 
-            if (foundEntry) {
-                setDate(foundEntry.date)
-                setWaterIntake(foundEntry.waterIntake)
-                setProteinIntake(foundEntry.proteinIntake)
-                setExercise(foundEntry.exercise)
-                setKegels(foundEntry.kegels)
-                setGarlandPose(foundEntry.garlandPose)
-                setPrenatalVitamins(foundEntry.prenatalVitamins)
-                setProbiotics(foundEntry.probiotics)
-                setTitle('Update Journal Entry')
-                setDescription('Update the form below and save changes to your journal entry.')
-            }
+            foundEntry = mockEntries.find(entry => entry.id === parseInt(entryId)) 
+        } 
+        if (foundEntry && user.id === foundEntry.userId) {
```

Comment:
Nice to not use nested conditional. Could the checking of the user.id matching could probably be added to the find? The backend could probably take two parameters that make sure the entry.id and user.id are correct.